currency/currency

```diff
@@ -136,14 +148,35 @@ checkInternet()
 ## Grabs the exchange rate and does the math for converting the currency
 convertCurrency()
```

Comment:
So if i am reading the code right, we are now making multiple queries for currencies that are not hard pegged to others? I feel like we need to make this special case behavior and not typical especially when its unnecessary.

Reply:
No, there's at most one query of https://api.exchangeratesapi.io, if that's what you mean.  
First, the code checks whether any of the input currencies is pegged to another currency:  
  
If yes, it grabs the hardcoded conversion rates, then contacts the API and then applies the hardcoded conversion rates as coefficients to the main conversion rate retrieved from the API.  
  
If not, the coefficient remain set to 1, so while the code still applies them to the main conversion rate, it has no effect.  
  
In the special case when we're converting between a pegged currency and its base currency, the API isn't contacted at all.

Reply:
Ah, I see the logic now, thank you for the explanation!